src/main/java/org/prebid/server/settings/CompositeApplicationSettings.java

```diff
@@ -0,0 +1,144 @@
+package org.prebid.server.settings;
+
+import io.vertx.core.Future;
+import org.prebid.server.execution.Timeout;
+import org.prebid.server.settings.model.Account;
+import org.prebid.server.settings.model.StoredRequestResult;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.BiFunction;
+
+/**
+ * Implements composite pattern for a list of {@link ApplicationSettings}.
+ */
+public class CompositeApplicationSettings implements ApplicationSettings {
+
+    private Proxy proxy;
+
+    public CompositeApplicationSettings(List<ApplicationSettings> delegates) {
+        if (Objects.requireNonNull(delegates).isEmpty()) {
+            throw new IllegalStateException("At least one application settings required. Please check configuration.");
```

Comment:
`IllegalArgumentException` would fit better in this case. Another thing to consider is to avoid config reference in the component - it's fine to leave something like this: "At least one application settings implementation required"